### ISSUE TITLE
Allow option for increasing quality of output video in `record`

### ISSUE THREAD

**Issue by Datseris** [MakieOrg/Makie.jl#2123]
At the moment, besides setting `compression = 1`, there is no way to further increase quality of video output of `record`, even when using CairoMakie backend. Even though that backend has several options for increasing the "crispiness", via `pt_per_unit` and `px_per_unit`, these options are not propagated into record.

It is possible to save individual images and then merge them with `ffmpeg` but it has downsides:
1. Very inconvenient or even impossible to do when videos come from packages establishing animations for Julia packages, such a video from Agents.jl
2. Some users simply do not have enough knowledge of `ffmpeg` to do this immediately.

**Comment by asinghvi17** [comment]
I wrote https://github.com/JuliaPlots/Makie.jl/pull/2012 to work with CairoMakie, I think it should work with all backends now but there is some debate on what should be the source of truth for the pixel size - `pixelarea(scene)` or `GeometryBasics.widths(screen)`.  Once that's understood I think this can be merged as is?

**Comment by Datseris** [comment]
`CairoMakie.activate!(px_per_unit=3)` is **exactly** what I wanted, but it doesn't work yet! Thanks for the PR!

**Comment by czimm79** [comment]
Hello! Datseris, do you have a workaround for this at the moment? I'd really like to make high-resolution animations in Makie.

**Comment by Datseris** [comment]
nope

**Comment by czimm79** [comment]
Thanks for the reply. I've written a workaround. This should be useful until #2012 is complete. As a bonus, it also allows me to get around another bug in Pluto.jl that prevents me from using `Makie.record`. fonsp/Pluto.jl#2283

I hope this can help others with the same problem for the time being.

```julia
"""
	brute_record(frame_func, path, iter; px_per_unit=3, fps=30)

Stitch together `Figure`'s output by `frame_func` into a video.

`frame_func` should return a `Figure` that can be saved as an image when supplied with an element of `iter`. 
Requires FFMPEG.jl.
"""
function brute_record(frame_func::Function, path::AbstractString, iter; px_per_unit=3, fps=30)
	
	# make temp dir to save images
	dir = mktempdir(prefix="brutemakie")
	
	for (idx, i) in pairs(iter)
		fig = frame_func(i)  # call function that creates a Figure
		save("$dir/$idx.png", fig; px_per_unit=px_per_unit)
	end

	# stitch images together
	FFMPEG.exe(`-f image2 -framerate $fps -i $dir/%d.png $path -y`)

	# remove the temp dir
	rm(dir, recursive=true)
end
```

## Example

```julia
brute_record("test.mp4", 0:0.1:2π) do x
	y = sin(x)
	fig = scatter([x], [y], axis=(;limits=(0, 7, -1.5, 1.5)))
end
```

https://user-images.githubusercontent.com/49537407/191623516-331a9d6b-f636-4a59-8c5a-c1d84ddc097a.mp4

**Comment by SimonDanisch** [comment]
```julia
brute_record("test.mp4", 0:0.1:2π) do x
	y = sin(x)
	fig = scatter([x], [y], axis=(;limits=(0, 7, -1.5, 1.5)))
end
```
That's really not how you should animate in Makie right now ;) I guess it's fine if it works for you, but don't be surprised if it's really slow ;) 

Let me see if I can merge https://github.com/MakieOrg/Makie.jl/pull/2012 today!

**Comment by Datseris** [comment]
(yeah I was going to modify that code to just call an in-place function that updates observables and then just re-save an existing figure)

**Comment by asinghvi17** [comment]
I believe this has been solved by #2306...
